Plugins/Event-Logs/Get-WindowsEvents.ps1

```diff
@@ -0,0 +1,287 @@
+
+    )
+
+#Application Event Log IDs to Retrieve (Application)
+$ApplicationEvents = @(
+
+        1000,  #Application errors and hangs (may be noisy)
+        1001,  #Application errors and hangs (may be noisy)
+        1002,  #Application errors and hangs (may be noisy)
+        1033,  #Installation completed (with success/failure status)
+        1034,  #Application removal completed (with success/failure status)
+        11707, #Installation completed successfully
+        11708, #Installation operation failed
+        11724  #Application removal completed successfully
+
+    )
+
+#Windows Firewall Event Log IDs to Retrieve (Microsoft-Windows-Windows Firewall With Advanced Security/Firewall.evtx)
+$WinFirewallEvents = @(
+
+        2003 #Windows firewall profile has been changed
+
+    )
+
+#PowerShell Event Log IDs to Retrieve (Microsoft-Windows-PowerShell/Operational.evtx)
+$PowerShellEvents = @(
+
+        4104, #Script contents
+        4105, #Script start
+        4106  #Script stop
+
+    )
+
+#WMI Event Log IDs to Retrieve (Microsoft-Windows-WMI-Activity/Operational.evtx)
+$WMI_Events = @(
+
+        5857, #Record filter/consumer activity
+        5858, #Record filter/consumer activity
+        5859, #Record filter/consumer activity
+        5860, #Record filter/consumer activity
+        5861  #New permanent event consumer creation
+
+    )
+
+#RDP Event Log IDs to Retrieve (Microsoft-Windows-TerminalServices-RDPClient/Operational.evtx)
+$RDP_TC_RDPClient_Events = @(
+
+        1024, #Destination Hostname (From Source/Initiating System - system RDP-ing from)
+        1102  #Destination IP Address (From Source/Initiating System - system RDP-ing from)
+
+)
+
+#RDP Event Log IDs to Retrieve (Microsoft-Windows-TerminalServices-RemoteConnectionManager/Operational.evtx)
+$RDP_TC_RCM_Events = @(
+
+        1149 #Source IP/Logon Username (On destination system-machine being RDP'd to)
+
+    )
+
+#RDP Event Log IDs to Retrieve (Microsoft-Windows-TerminalServices-LocalSessionManager/Operational.evtx)
+$RDP_TC_LSM_Events = @(
+
+        21, #Source IP/Logon Username
+        22, #Source IP/Logon Username
+        25, #Source IP/Logon Username
+        41  #Source IP/Logon Username
+    )
+
+#RDP Event Log IDs to Retrieve (Microsoft-Windows-RemoteDesktopServices-RdpCoreTS/Operational.evtx)
+$RDP_RdpTS_Events = @(
+
+        98, #Successful Connections
+        131 #Connection Attempts (Source IP/Logon UserName)
+
+    )
+
+#Scheduled Tasks Event Log IDs to Retireve (Microsoft-Windows-Task Scheduler/Operational.evtx)
+$Sched_Tasks_Events = @(
+
+        106, #Scheluded task created
+        140, #Scheduled task updated
+        141, #Scheduled task deleted
+        200, #Scheduled task executed
+        201  #Scheduled task completed
+
+    )
+
+#if no date is supplied in a parameter through Power-Response, fill in start date based on number of days of assumed dwell time
+
+if ($Start_Date.length -gt 1) {
```

Comment:
Rather than having this section to convert the `[String]$Start_Date` to `[DateTime]$Date`, you could make the `$Start_Date` parameter of type `[DateTime]` and give it a default value of `(Get-Date).AddDays(-90)`